ast/compile.go

```diff
@@ -91,6 +91,10 @@ type Compiler struct {
 	pathExists func([]string) (bool, error)
 	after      map[string][]CompilerStageDefinition
 	metrics    metrics.Metrics
+
+	// This is a set of unsafe built-in functions that we are not allowed to
+	// use.
+	unsafeBuiltinsMap map[string]bool
```

Comment:
Nit: I prefer to use `map[string]struct{}` in place of `map[string]bool` when representing sets since there is less opportunity for misinterpreting the value.